root/tests/Feature/ExampleTest.php

```diff
@@ -2,20 +2,83 @@
 
 namespace Tests\Feature;
 
-// use Illuminate\Foundation\Testing\RefreshDatabase;
+use Illuminate\Foundation\Testing\RefreshDatabase;
 use Tests\TestCase;
+use App\Http\Controllers\HousingLoanChartController;
 
 class ExampleTest extends TestCase
 {
+    public function testPostPage2()
+    {
+
+        // テスト用のリクエストデータを作成
+        $requestData = [
+            'テスト'
+        ];
+
+        // リクエストを送信
+        $response = $this->post('housing-loan/question-page/page2', $requestData);
+
+        // レスポンスを検証
+        $response->assertRedirect('/housing-loan/question-page/page3');
+
+        // セッションにデータが正しく保存されたかを検証
+        $this->assertEquals($requestData, session('form.page2'));
+    }
+
+    public function nameProvider()
+    {
+        return [
+            [1, 1, 1, 1, 1, "anna"],
+            [1, 0, 0, 0, 0, "anna"],
+            [1, 1, 0, 0, 0, "anna"],
+            [1, 0, 1, 0, 0, "anna"],
+            [1, 0, 0, 1, 0, "anna"],
+            [1, 0, 0, 0, 1, "anna"],
+            [2, 1, 1, 1, 1, "anna"],
+            [2, 0, 0, 0, 0, "anna"],
+            [2, 1, 0, 0, 0, "anna"],
+            [2, 0, 1, 0, 0, "anna"],
+            [2, 0, 0, 1, 0, "anna"],
+            [2, 0, 0, 0, 1, "anna"],
+            [3, 0, 0, 0, 0, "anna"],
```

Comment:
[IMO]
financial_institutionはbooleanで入れ子になる可能性が高いので
ある程度パターンの網羅が必要ですが
基本的には関連性のある項目はないので、項目一つ一つでパターンを作ると良いです。


```suggestion
            'Case01: inancial_institution All 1' => [3, 1, 1, 1, 1, "anna"],
            'Case02: inancial_institution All 0' => [3, 0, 0, 0, 0, "anna"],
            'Case03: inancial_institution1 = 1' => [3, 1, 0, 0, 0, "anna"],
            'Case04: inancial_institution2 = 1' => [3, 0, 1, 0, 0, "anna"],
            'Case05: inancial_institution3 = 1' => [3, 0, 0, 1, 0, "anna"],
            'Case06: inancial_institution4 = 1' => [3, 0, 0, 0, 1, '日本語・記号・数値・あa1!?#'],
            'Case07: usage_situation = 2, name min length' => [2, 0, 0, 0, 0, 'a'],
            'Case08: usage_situation = 1,  name max length' => [1, 0, 0, 0, 0, str_repeat('a', 255)],
```

Reply:
テストコードの例を作成いただきありがとうございます！
テストケース作成の知識がまだまだ浅いので、勉強します。

上記のようにコード修正しました。
[7a5cb2e](https://github.com/anna-nouchi-epkotsoftware/survey-site-laravel/pull/2/commits/7a5cb2e4bbd9cad12a288d44a0e71a91d01a5104)